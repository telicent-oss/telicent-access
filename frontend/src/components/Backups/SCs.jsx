import React from 'react'
import config from '../../config/app-config'
import SCBackup from './SCBackup'

const smartCaches = [
    {
        name: "Smart Cache Graph",
        identifier: "scg",
        url: config.sparql.url,
        type: "triplestore"
    }, {
        name: "Smart Cache Search",
        identifier: "scs",
        url: config.search.url,
        type: "search index"
    }
]
const SCs = () => {
    return (
        <ul className='px-10 m-4'>
            {smartCaches.map(sc => <SCBackup key={sc.identifier} name={sc.name} identifier={sc.identifier} url={sc.url} type={sc.type} />)}
        </ul>
    )
}
export default SCs;