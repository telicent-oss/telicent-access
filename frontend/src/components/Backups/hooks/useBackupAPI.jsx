import { useState } from "react";
import { backupAPI } from "../api";

// To check tomorrow -> what is the plan re: API, are failures delivered as a 200?
const useBackupAPI = (url, identifier) => {
  const api = backupAPI[identifier];
  const [loading, setLoading] = useState(false);

  const listBackups = async () => {
    try {
      setLoading({ message: "Fetching a list of backups" });
      const data = await api.getList(url);
      return [data, null];
    } catch (e) {
      // eslint-disable-next-line no-console
      console.log(e.message);
      return [[], "Failed to retrieve backups"];
    } finally {
      setLoading(null);
    }
  };

  const deleteSingleBackup = async (id) => {
    try {
      setLoading({ message: `Deleting backup: ${id}` });
      const data = await api.delete(url, id);
      return [data, null];
    } catch (e) {
      // eslint-disable-next-line no-console
      console.log(e.message);
      return [null, `Failed to delete backup: ${id}`];
    } finally {
      setLoading(null);
    }
  };

  const createBackup = async () => {
    try {
      setLoading({ message: "Creating new backup" });
      const data = await api.create(url);
      return [data, null];
    } catch (e) {
      // eslint-disable-next-line no-console
      console.log(e.message);
      return [null, "Failed to create backup"];
    } finally {
      setLoading(null);
    }
  };
  const restoreSingleBackup = async (id) => {
    try {
      setLoading({ message: `Restoring backup: ${id}` });
      const data = await api.restore(url, id);
      return [data, null];
    } catch (e) {
      // eslint-disable-next-line no-console
      console.log(e.message);
      return [null, `Failed to restore backup: ${id}`];
    } finally {
      setLoading(null);
    }
  };

  return {
    loading,
    listBackups,
    deleteSingleBackup,
    restoreSingleBackup,
    createBackup,
  };
};

export default useBackupAPI;
