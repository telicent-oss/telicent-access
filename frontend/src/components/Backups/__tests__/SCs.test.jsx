/* eslint-disable import/first */
jest.mock("../hooks/useBackupAPI.jsx", () => ({
  __esModule: true,
  default: jest.fn()
}));

import { render } from "@testing-library/react"
import SCs from "../SCs"
import useBackupAPI from "../hooks/useBackupAPI";

describe("SCs", () => {

  beforeEach(() => {
    useBackupAPI.mockReturnValue({
      loading: false,
      listBackups: jest.fn().mockResolvedValue([[], null]),
      deleteSingleBackup: jest.fn(),
      restoreSingleBackup: jest.fn(),
      createBackup: jest.fn()
    });
  });

  it("should rendenr", () => {
    const { container } = render(<SCs />)
    expect(container).toMatchSnapshot();
  })
})
