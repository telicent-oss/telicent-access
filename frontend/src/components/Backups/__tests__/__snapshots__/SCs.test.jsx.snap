// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCs should rendenr 1`] = `
<div>
  <ul
    class="px-10 m-4"
  >
    <div
      class="p-4 bg-black-200 rounded-lg mb-8 "
    >
      <div
        class="flex justify-between mb-4 w-full"
      >
        <div
          class=" flex "
        >
          <i
            class="fa-solid fa-database m-4 text-xl"
          />
          <div>
            <h1
              class="text-2xl"
            >
              Smart Cache Graph
            </h1>
            <small>
              triplestore
            </small>
          </div>
        </div>
        <div
          class="flex mr-4"
        >
           
          <button
            class="MuiButtonBase-root teli-button teli-button--small teli-button__secondary true h-[34.5px] css-10d1a0h-MuiButtonBase-root"
            id="create-backup"
            tabindex="0"
            type="button"
          >
            Create
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="mx-12 flex items-center"
      >
        <p>
          No backups available
        </p>
      </div>
    </div>
  </ul>
</div>
`;
